go/carmen/database.go

```diff
@@ -40,13 +41,24 @@ func openDatabase(
 	if err != nil {
 		return nil, err
 	}
+	checkpointInterval, err := properties.GetInteger(CheckpointInterval, 0)
+	if err != nil {
+		return nil, err
+	}
+	checkpointPeriod, err := properties.GetInteger(CheckpointPeriod, 0)
+	if err != nil {
+		return nil, err
+	}
+
 	params := state.Parameters{
-		Directory:    directory,
-		Variant:      state.Variant(configuration.Variant),
-		Schema:       state.Schema(configuration.Schema),
-		Archive:      state.ArchiveType(configuration.Archive),
-		LiveCache:    int64(liveCache),
-		ArchiveCache: int64(archiveCache),
+		Directory:          directory,
+		Variant:            state.Variant(configuration.Variant),
+		Schema:             state.Schema(configuration.Schema),
+		Archive:            state.ArchiveType(configuration.Archive),
+		LiveCache:          int64(liveCache),
+		ArchiveCache:       int64(archiveCache),
+		CheckpointInterval: checkpointInterval,
+		CheckpointPeriod:   time.Duration(checkpointPeriod),
```

Comment:
I would assume that this is interpreting the given `checkpointPeriod` in nanoseconds (see definition of type [time.Duration](https://pkg.go.dev/time#Duration)).

This should probably be something like:
```suggestion
		CheckpointPeriod:   checkpointPeriod * time.Minute,
```